pymongo/bulk.py

```diff
@@ -393,16 +423,16 @@ def execute_command_no_results(self, sock_info, generator):
         # Ordered bulk writes have to be acknowledged so that we stop
         # processing at the first error, even when the application
         # specified unacknowledged writeConcern.
-        write_concern = WriteConcern()
+        initial_write_concern = WriteConcern()
         op_id = _randint()
         try:
             self._execute_command(
-                generator, write_concern, None,
-                sock_info, op_id, False, full_result)
+                generator, initial_write_concern, None,
+                sock_info, op_id, False, full_result, write_concern)
         except OperationFailure:
             pass
 
-    def execute_no_results(self, sock_info, generator):
+    def execute_no_results(self, sock_info, generator, write_concern=None):
```

Comment:
Let's make write_concern a required argument since we always pass it.

Reply:
Done